### PR TITLE
Bookstore : Create remove books

### PR DESCRIPTION
**Pull request by oliverSCZ**
I did the following:

- Created a Repo and a branch called initialize_project
- Used correct Gitflow
- Effectively set up a React-Redux linter.
- Created basic setup to work with React.js
- Setup the Redux Store
- Added new books to set the data inputs in the local React state
- In my React component responsible for removing books - implemented that even by dispatching a corresponding action.
- Tested in the browser

### CODE REVIEW

File: src/components/AddNewBookForm.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        name="author"
+        value={author}
+        type="text"
+        placeholder="Author"
+        onChange={inputChange}
+      />
+      <button type="submit">Add Book</button>
+    </form>
+  );
+};
```
Comment (Emmyn5600) [inline src/components/AddNewBookForm.js]: - [x] Let us prevent a user from adding the empty data, let us validate our input fields.

File: src/index.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 
 ReactDOM.render(
   <React.StrictMode>
-    <Router>
-      <App />
-    </Router>
+    <Provider store={store}>
+      <Router>
+        <App />
+      </Router>
```
Comment (Emmyn5600) [inline src/index.js:14]: - [x] You have used the routes in the app.js file, no need to wrap the app here again.

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    case ADD_BOOK:
+      return [...state, action.payload];
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.payload);
+    default:
+      return state;
+  }
+};
+
+export default reducer;
```
Comment (Emmyn5600) [inline src/redux/books/books.js:27]: - [ ] Optional suggestions

Good job here, however, I would recommend separating the action type, creator and reducers into different files to keep our code clean and easy to manage.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @oliverSCZ 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.

 ## Required Changes ♻️
- Check the comments under the review.

## Optional suggestions

- n/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/index.js (hunk 2 above)
Comment (oliverSCZ) [reply src/index.js:14]: I'm sorry but, it doesn't work if I take it out. I tried.

Comment (Emmyn5600) [reply src/index.js:14]: You can leave it now but try to find out the way it should be solved.

File: src/components/AddNewBookForm.js (hunk 1 above)
Comment (rloterh) [reply src/components/AddNewBookForm.js]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Oliver, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**